docs/sources/github.md

```diff
@@ -0,0 +1,47 @@
+# Event Source for Github
+
+This event source creates a webhook to listen for incoming Github Events, turning received requests into [CloudEvents][ce] to be consumed by other TriggerMesh components.
+
+## Prerequisite(s)
+
+- GitHub Tokens
+
+### Create GitHub Tokens
+Create a personal access token for GitHub that the GitHub source can use to register webhooks with the GitHub API. Also decide on a secret token that your code will use to authenticate the incoming webhooks from GitHub (secretToken).
+
+The token can be named anything you find convenient. The Source requires repo:public_repo and admin:repo_hook, to let it fire events from your public repositories and to create webhooks for those repositories. Copy and save this token; GitHub will force you to generate it again if misplaced.
```

Comment:
How about annotating the oauth permissions `repo:public_repo` and `admin:repo_hook`? Also, are there any additional permissions required for private repos as well?

Reply:
I dont. know. I pulled this directly from here -> https://knative.dev/docs/eventing/samples/github-source/